libvast/src/system/active_partition.cpp

```diff
@@ -329,7 +323,7 @@ active_partition_actor::behavior_type active_partition(
   active_partition_actor::stateful_pointer<active_partition_state> self,
   uuid id, accountant_actor accountant, filesystem_actor filesystem,
   caf::settings index_opts, const index_config& synopsis_opts,
-  store_actor store, std::string store_id, chunk_ptr header,
+  const store_actor_plugin* store_plugin,
```

Comment:
I think in terms of API it'd be nicer to just pass in the name of the store plugin here rather than doing the lookup in the index actor, as then the index actor doesn't need to know about the store at all.

Reply:
I'd agree if we wanted to stick with the index as it is, but we're going to replace it with a "partition builder" in the medium term, and that can definitely do the plugin lookup on behalf of the partitions it creates.

Reply:
I still think that'd be cleaner code, but I don't want to argue about this—as you said, this is bound to be replaced soon regardless.